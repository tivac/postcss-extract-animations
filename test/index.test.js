"use strict";

var dedent = require("dedent"),

    plugin = require("../index.js");

describe("postcss-extract-animations", () => {
    it("should extract animations from simple selectors", () =>
        plugin.process(dedent(`
            .a {
                animation: 1s @keyframes {
                    to {
                        opacity: 1;
                    }
                }
            }
        `))
        .then((result) => expect(result.css).toMatchSnapshot())
    );

    it("should extract animations from more complex selectors", () =>
        plugin.process(dedent(`
            .a .b > div > a[animate] {
                animation: 1s @keyframes {
                    to {
                        opacity: 1;
                    }
                }
            }
        `))
        .then((result) => expect(result.css).toMatchSnapshot())
    );

    it("should extract animations from more complex declarations", () =>
        plugin.process(dedent(`
            .a {
                animation: 1s ease-in 1s 2 reverse both paused @keyframes {
                    to {
                        opacity: 1;
                    }
                }
            }
        `))
        .then((result) => expect(result.css).toMatchSnapshot())
    );


    it("shouldn't touch declarations around the animation", () =>
        plugin.process(dedent(`
            .a {
                color: #FFF;
                animation: 1s @keyframes {
                    to {
                        opacity: 1;
                    }
                }
                background: red;
            }
        `))
        .then((result) => expect(result.css).toMatchSnapshot())
    );

    it("shouldn't touch normal animation decls", () =>
        plugin.process(dedent(`
            .a {
                animation: 1s reveal;
                animation: 1s @keyframes {
                    to {
                        opacity: 1;
                    }
                }
            }
        `))
        .then((result) => expect(result.css).toMatchSnapshot())
    );
});
