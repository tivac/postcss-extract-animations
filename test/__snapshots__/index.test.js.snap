// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`postcss-extract-animations should extract animations from more complex declarations 1`] = `
"@keyframes anim-c4fe818f {
        to {
            opacity: 1;
        }
}
.a {
    animation: 1s ease-in 1s 2 reverse both paused anim-c4fe818f
}"
`;

exports[`postcss-extract-animations should extract animations from more complex selectors 1`] = `
"@keyframes anim-7947fab3 {
        to {
            opacity: 1;
        }
}
.a .b > div > a[animate] {
    animation: 1s anim-7947fab3
}"
`;

exports[`postcss-extract-animations should extract animations from simple selectors 1`] = `
"@keyframes anim-c4fe818f {
        to {
            opacity: 1;
        }
}
.a {
    animation: 1s anim-c4fe818f
}"
`;

exports[`postcss-extract-animations shouldn't touch declarations around the animation 1`] = `
"@keyframes anim-c4fe818f {
        to {
            opacity: 1;
        }
}
.a {
    color: #FFF;
    animation: 1s anim-c4fe818f;
    background: red;
}"
`;

exports[`postcss-extract-animations shouldn't touch normal animation decls 1`] = `
"@keyframes anim-c4fe818f {
        to {
            opacity: 1;
        }
}
.a {
    animation: 1s reveal;
    animation: 1s anim-c4fe818f
}"
`;
